.github/workflows/docker.yml

```diff
@@ -0,0 +1,187 @@
+
+      - name: Compute short hash
+        id: short_hash
+        run: |
+          SHORT_HASH=$(git rev-parse --short ${{ github.sha }})
+          echo "SHORT_HASH=${SHORT_HASH}" >>${GITHUB_OUTPUT}
+
+      - name: Lowercase repository owner
+        id: lowercase_repo_owner
+        run: |
+          echo "REPO_OWNER=${REPO_OWNER,,}" >>${GITHUB_OUTPUT}
+        env:
+          REPO_OWNER: "${{ github.repository_owner }}"
+
+      - name: Scoped board server sqlite image name
+        id: scoped_board_server_sqlite_image_name
+        run: |
+          IMAGE_NAME=${{ steps.lowercase_repo_owner.outputs.REPO_OWNER }}/${{ env.BOARD_SERVER_SQLITE_IMAGE_NAME }}
+          echo "IMAGE_NAME=${IMAGE_NAME}" >>${GITHUB_OUTPUT}
+
+      - name: Board server sqlite image path
+        id: board_server_sqlite_image_path
+        run: |
+          echo "IMAGE_PATH=${{ env.REGISTRY }}/${{ steps.scoped_board_server_sqlite_image_name.outputs.IMAGE_NAME }}" >>${GITHUB_OUTPUT}
+
+      - name: Board server firestore image name
+        id: board_server_firestore_image_name
+        run: |
+          echo "IMAGE_NAME=${{ steps.lowercase_repo_owner.outputs.REPO_OWNER }}/${{ env.BOARD_SERVER_FIRESTORE_IMAGE_NAME }}" >>${GITHUB_OUTPUT}
+
+      - name: Board server firestore image path
+        id: board_server_firestore_image_path
+        run: |
+          echo "IMAGE_PATH=${{ env.REGISTRY }}/${{ steps.board_server_firestore_image_name.outputs.IMAGE_NAME }}" >>${GITHUB_OUTPUT}
+
+      - name: Get package version
+        id: package_version
+        run: |
+          VERSION=$(node -p "require('./packages/board-server/package.json').version")
+          echo "VERSION=${VERSION}" >> ${GITHUB_OUTPUT}
+
+    outputs:
+      repo_owner: ${{ steps.lowercase_repo_owner.outputs.REPO_OWNER }}
+      short_hash: ${{ steps.short_hash.outputs.SHORT_HASH }}
+
+      board_server_sqlite_image_name: ${{ steps.scoped_board_server_sqlite_image_name.outputs.IMAGE_NAME }}
+      board_server_sqlite_image_path: ${{ steps.board_server_sqlite_image_path.outputs.IMAGE_PATH }}
+
+      board_server_firestore_image_name: ${{ steps.board_server_firestore_image_name.outputs.IMAGE_NAME }}
+      board_server_firestore_image_path: ${{ steps.board_server_firestore_image_path.outputs.IMAGE_PATH }}
+
+      package_version: ${{ steps.package_version.outputs.VERSION }}
+
+  board-server-sqlite-image:
+    name: Build board server sqlite image
+    needs:
+      - setup
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Registry Auth
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Note: this will need to be set up by someone with the appropriate access.

Reply:
I updated the `GITHUB_TOKEN` permissions to look like this:

<img width="766" alt="image" src="https://github.com/user-attachments/assets/9269edc2-a72a-46b4-9f85-4230d0396872">

If I am reading the docs correctly, it sounds like we should now add a permissions section to all of our other workflows that remove any permissions we don't need, since by changing this all our workflows now have elevated write permissions by default. Is that your understanding too?

Reply:
No, I don't think that's the case.. see [here](https://github.com/breadboard-ai/breadboard/actions/runs/10617115498/job/29429132562) for example, this was a normal workflow run triggered on this branch and if you expand "Set up Job" and then expand the token permissions, they are still all read. I'll have another look at the docs.

Reply:
I just re-ran some arbitrary job since I changed this, and the permissions did seem to change now that I've updated the default:

Before:
```
GITHUB_TOKEN Permissions
  Contents: read
  Metadata: read
  Packages: read
```

After:
```
GITHUB_TOKEN Permissions
  Actions: read
  Attestations: read
  Checks: read
  Contents: read
  Deployments: read
  Discussions: read
  Issues: read
  Metadata: read
  Packages: read
  Pages: read
  PullRequests: read
  RepositoryProjects: read
  SecurityEvents: read
  Statuses: read
```

I don't see any writes there, though, so maybe it's fine. Also, it could just be that I launched this one manually. Let's just land this and see what happens with the next few workflows. We can pare down any permissions if needed as a followup.

Reply:
Alright let's do it 🚀